### PR TITLE
Make automation editor card headers translateable (triggers partly) 

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Start introducing translations for some of the triggers. Other triggers will get an individual PR. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: #13820
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/automation_i18n.ts [hunk 1]
```diff
@@ -101,14 +104,24 @@ export const describeTrigger = (
     }
 
     const eventTypesString = disjunctionFormatter.format(eventTypes);
-    return `When ${eventTypesString} event is fired`;
+    return hass.localize(
+      `${triggerTranslationBaseKey}.event.description.full`,
+      "eventTypes",
+      eventTypesString
```
Comment (bramkragten) [inline src/data/automation_i18n.ts:110]:
```suggestion
      {eventTypes: eventTypesString}
```

Comment (bramkragten) [reply src/data/automation_i18n.ts:110]: Please pass an object with placeholders (applies to all)

File: src/translations/en.json [hunk 2]
```diff
@@ -2308,12 +2311,20 @@
                   "label": "Home Assistant",
                   "event": "Event:",
                   "start": "Start",
-                  "shutdown": "Shutdown"
+                  "shutdown": "Shutdown",
+                  "description": {
+                    "full": "When Home Assistant is {startedOrShutdown}",
+                    "started": "started",
+                    "shutdown": "shutdown"
```
Comment (bramkragten) [inline src/translations/en.json:2318]: I'm not sure if we can do this. maybe the order of the sentence is completely different in some languages for started and shutdown for example.

File: src/translations/en.json [hunk 3]
```diff
@@ -2330,22 +2341,38 @@
                   "event": "[%key:ui::panel::config::automation::editor::triggers::type::homeassistant::event%]",
                   "sunrise": "Sunrise",
                   "sunset": "Sunset",
-                  "offset": "Offset (optional)"
+                  "offset": "Offset (optional)",
+                  "description": {
+                    "full": "When the sun {setsOrRises} {optionalOffset}",
```
Comment (bramkragten) [inline src/translations/en.json:2346]:
```suggestion
                    "full": "When the sun {setsOrRises}{optionalOffset}",
```

File: src/translations/en.json [hunk 4]
```diff
@@ -2330,22 +2341,38 @@
...
+                    "sets": "sets",
+                    "rises": "rises",
+                    "offset": "offset by {offset}"
```
Comment (bramkragten) [inline src/translations/en.json:2349]:
```suggestion
                    "offset": " offset by {offset}"
```

**Comment by silamon** [comment]
It might be better to use https://formatjs.io/docs/core-concepts/icu-syntax/#select-format where applicable instead of multiple translation keys for different parts of the sentence.

**Comment by bramkragten** [comment]
> It might be better to use [formatjs.io/docs/core-concepts/icu-syntax/#select-format](https://formatjs.io/docs/core-concepts/icu-syntax/#select-format) where applicable instead of multiple translation keys for different parts of the sentence.

That's pretty neat yeah 👍

**Review by bramkragten — APPROVED** [review]

**Pull request merged**